web/blocks.html

```diff
@@ -17,101 +17,173 @@
+              id="saveAsButton"
+              class="cw-button blue"
+              onclick="saveProjectAs()"
+            >
+              <i class="mdi mdi-18px mdi-dots-horizontal"></i>&nbsp; Save As
+            </a>
+            <a
+              id="deleteButton"
+              class="cw-button red"
+              style="display: none;"
+              onclick="deleteProject()"
+            >
+              <i class="mdi mdi-18px mdi-delete"></i>&nbsp; Delete
+            </a>
+          </div>
+          <div id="directoryTree"></div>
+        </div>
 
-            <div class="dropbox ui-layout-center" id="blocklyDiv" style="height: 84%; width:100%;">
-            </div>
+        <div
+          class="dropbox ui-layout-center"
+          id="blocklyDiv"
+          style="height: 84%; width: 100%;"
+        ></div>
 
-            <div id="result" style="display:none" class="ui-layout-east">
-                <div id="obsolete-code-alert" class="obsolete-code-alert-fadeout"></div>
-                <iframe id="runner" class="dropbox" width="500" height="500" style="display:none;"></iframe>
+        <div id="result" style="display: none;" class="ui-layout-east">
+          <div
+            id="obsolete-code-alert"
+            class="obsolete-code-alert-fadeout"
+          ></div>
+          <iframe
+            id="runner"
+            class="dropbox"
+            width="500"
+            height="500"
+            style="display: none;"
+          ></iframe>
 
-                <pre id="message" class="dropbox" style="width:100%; min-height:100px;
-                                                             padding:5px;display:none;"></pre>
+          <pre
+            id="message"
+            class="dropbox"
+            style="width: 100%; min-height: 100px; padding: 5px; display: none;"
+          ></pre>
 
-                <span><a type="button" style="margin-top:5px;" id="editButton" target="_blank" class="cw-button blue">
-                        <span>Edit code</span>
-                    </a>
-                </span>
-                <pre id="genCode" class="dropbox cm-s-default CodeMirror" style="width:100%; height:100%; padding:10px;">
-                    </pre>
-            </div>
+          <span
+            ><a
+              type="button"
+              style="margin-top: 5px;"
+              id="editButton"
+              target="_blank"
+              class="cw-button blue"
+            >
+              <span>Edit code</span>
+            </a>
+          </span>
+          <pre
+            id="genCode"
+            class="dropbox cm-s-default CodeMirror"
+            style="width: 100%; height: 100%; padding: 10px;"
+          ></pre>
+        </div>
+      </div>
+      <div id="toolbar">
+        <div id="buttons">
+          <a
+            id="navButton"
+            class="cw-button blue"
+            onclick="window.mainLayout.toggle('west')"
+            ><i class="mdi mdi-18px mdi-menu"></i
+          ></a>
```

Comment:
What happened to `</i>` here?  Frankly, this is really ugly.  The tag didn't even start at that indent level.  :(  Can it be turned off?

Same below.

Reply:
Wow. That's bizarre. I agree with you that it's really ugly. Should've been:

```html
<a
  id="navButton"
  class="cw-button blue"
  onclick="window.mainLayout.toggle('west')"
>
  <i class="mdi mdi-18px mdi-menu"></i>
</a>
```

I'll look into it.

Reply:
Looking at this again, I suspect the reason for that formatting is that adding whitespace between tags is actually a semantic change to the document, while whitespace inside a tag is ignored by the parser.  I've seen this make a difference before.  js-beautify did something similar with string interpolation in JavaScript, changing this:

```
    let a = "this is a really long string " + someExpr()
        + " and some more really really long string";
```

into this:

```
    let a = `this is a really long string ${someExpr()
        } and some more really really long string`;
```

I bet, in this case, if you added a space into the source document, prettier would stop breaking the line inside the tag.

Reply:
I think you are on point about whitespace sensitivity =>

https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting

Reply:
So this isn't a big deal.  When the whitespace doesn't hurt, we can add it by hand.